test/native/helpers.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react-native';
+
+export async function initializeEditor( { initialHtml } ) {
+	const Editor = require( '@wordpress/edit-post/src/editor' ).default;
+	const renderResult = render(
+		<Editor
+			postId={ 0 }
+			postType="post"
+			initialTitle="test"
+			initialHtml={ initialHtml }
+		/>
```

Comment:
Trying to follow what we do in UI tests, I'm rendering the editor component with an initial HTML. This would be the same as what we do when the editor is initialized:

https://github.com/WordPress/gutenberg/blob/f3f931bd68efa0e60e09a9b8d0e7dd2464c05041/packages/react-native-editor/src/index.js#L47-L48

https://github.com/WordPress/gutenberg/blob/f3f931bd68efa0e60e09a9b8d0e7dd2464c05041/packages/edit-post/src/index.native.js#L24-L32

The initial props that are streamlined from the native side are included here when the editor is initialized:
https://github.com/WordPress/gutenberg/blob/f3f931bd68efa0e60e09a9b8d0e7dd2464c05041/packages/element/src/react-platform.native.js#L19-L30

In the case of the integration tests, we can directly add them in the render call.

Reply:
I considered this more and I believe there are likely three approaches for rendering the required tree for the subject test.

1. Manually build the React tree. 
1. Tap the ➕ (Add block) button and select the block. 
1. Render the `Editor` with the relevant initial HTML. 

The first approach is what the Cover edit tests do. It makes some presumptions about the "testing environment" and builds a tree it requires/expects. The benefit may be that it is arguably simpler to understand from within the test file, but its not testing as much functionality. 

The second approach fully tests a typical clean-slate flow from a user's perspective, but requires rendering a lot more UI, getting us closer to e2e tests. 

The third approach tests editing existing block HTML, as the editor is required to parse the initial HTML to render the relevant UI. The negative here being that one has to create HTML fixtures for the tests.  

Out of those three, it does seem like the third is a good balance of manageability and delivering value. So, I lean towards following the approach you have outlined here in your PR. WDYT?

Reply:
I'm also leaning on the third option, we would have full functionality and it's going to be the closest environment to what we do in the UI tests and the user experience.

One of the key points I think we could deliver with this type of tests is the quick identification of breakages, if we test the full flow of the editor we probably will spot most of them 🔧 .